### PR TITLE
feat: Facebook delivery reports

### PR DESCRIPTION
**Pull request by muhsin-k**
Fixes https://linear.app/chatwoot/issue/CW-2570/delivery-report-for-facebook

### CODE REVIEW

File: config/initializers/facebook_messenger.rb [hunk 1]
```diff
@@ -29,14 +29,13 @@ def bot
...
-    # delivery.seq       # => 37
-    updater = Integrations::Facebook::DeliveryStatus.new(delivery)
-    updater.perform
-    Rails.logger.info "Human was online at #{delivery.at}"
+    Rails.logger.info "Recieved delivery status #{delivery.to_json}"
+    Webhooks::FacebookDeliveryJob.perform_later(delivery.to_json)
+  end
+
+  Facebook::Messenger::Bot.on :read do |read|
+    Rails.logger.info "Recieved read status  #{read.to_json}"
```
Comment (muhsin-k) [inline config/initializers/facebook_messenger.rb:37]: Added these logs to verify if events are being received in production, as they are not appearing locally.

File: app/jobs/webhooks/facebook_delivery_job.rb
Comment (scmmishra) [inline app/jobs/webhooks/facebook_delivery_job.rb:1]: This seems fine, perhaps in the future we might need to add the mutex here, but got to ensure that the sequence of events stay the same before that. No need to change anything for now

File: lib/integrations/facebook/delivery_status.rb [hunk 2]
```diff
@@ -1,32 +1,37 @@
...
+    process_delivery_status if params.delivery_watermark
+    process_read_status if params.read_watermark
   end
 
   private
 
-  def sender_id
-    @params.sender['id']
+  def process_delivery_status
+    timestamp = Time.zone.at(params.delivery_watermark.to_i).to_datetime
```
Comment (scmmishra) [inline lib/integrations/facebook/delivery_status.rb:17]: Can you confirm if this saves the time in UTC 

Cc: @sojan-official can you confirm if this is required

Comment (muhsin-k) [reply lib/integrations/facebook/delivery_status.rb:17]: I believe `Time.zone` defaults to UTC.  The payload is something like this,
CleanShot 2023-11-16 at 13 08 49@2x

Comment (sojan-official) [reply lib/integrations/facebook/delivery_status.rb:17]: https://chat.openai.com/share/781e05cf-afab-488d-afb3-4885e7df3d82

Comment (muhsin-k) [reply lib/integrations/facebook/delivery_status.rb]: @sojan-official @scmmishra Fixed.

**Review by scmmishra — APPROVED** [review]

**Pull request merged**